src/hooks/useReferralCode.ts

```diff
@@ -37,6 +39,7 @@ export function useReferralCode() {
     if (referredBy?.affiliateAddress) return;
 
     if (referralAddress) {
+      track(AnalyticsEvents.AffiliateSaveReferralAddress({ affiliateAddress: referralAddress }));
```

Comment:
will this occasionally have false positives since we have the side effect to removeLatestReferrer?

Reply:
I think we should actually add tracking to removing the referrer and add a string as the reason.

Reply:
we can combine this event + not removing referrer to get a more accurate count

Reply:
That makes sense! I'll update